libminifi/src/core/ProcessGroup.cpp

```diff
@@ -403,6 +403,16 @@ void ProcessGroup::removeConnection(std::shared_ptr<Connection> connection) {
   }
 }
 
+void ProcessGroup::drainConnections() {
+  for (auto &&connection : connections_) {
+    connection->drain();
+  }
+
+  for (std::set<ProcessGroup *>::iterator it = child_process_groups_.begin(); it != child_process_groups_.end(); ++it) {
+    (*it)->drainConnections();
+  }
```

Comment:
This could become a range-based for loop.

Reply:
done